CSS counter() function (#30583)
mdn/content: files/en-us/web/css/counter/index.md

{{CSSRef}}

The **`counter()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Functions) returns a string representing the current value of the named counter, if there is one.

> The `counter()` function is generally used within [pseudo-element](/en-US/docs/Web/CSS/Pseudo-elements) {{CSSxRef("content")}} property values, but can theoretically be used anywhere a [`<string>`](/en-US/docs/Web/CSS/string) value is supported.

Reviewer: Please apply the same edit here as suggested in counters() file

> The `counter()` function is generally used within [pseudo-element](/en-US/docs/Web/CSS/Pseudo-elements) {{CSSxRef("content")}} property values, but can theoretically be used anywhere a [`<string>`](/en-US/docs/Web/CSS/string) value is supported.

Author: ```suggestion
The `counter()` function is generally used within [pseudo-element](/en-US/docs/Web/CSS/Pseudo-elements) through the {{cssxref("content")}} property but, theoretically, it can be used wherever a [`<string>`](/en-US/docs/Web/CSS/string) value is supported.
```

The `counter()` and {{cssxref("counters", "counters()")}} functions are what make counters useful by returning developer-defined strings (or images).

### Values

The `counter()` function accepts up to two parameters. The first parameter is the `<counter-name>`. The optional second parameter is the `<counter-style>`.

- `<counter-name>`
  - : A {{cssxref("&lt;custom-ident&gt;")}} identifying the counter, which is the same case-sensitive name used with the {{cssxref("counter-reset")}} and {{cssxref("counter-increment")}} property values. The counter name cannot start with two dashes and can't be `none`, `unset`, `initial`, or `inherit`.
- `<counter-style>`
  - : A {{cssxref("&lt;list-style-type&gt;")}} name, {{cssxref("&lt;@counter-style&gt;")}} name or {{cssxref("symbols", "symbols()")}} function, where a counter style name is a `numeric`, `alphabetic`, or `symbolic` simple predefined counter style, a complex longhand east Asian or Ethiopic predefined counter style, or other [predefined counter style](/en-US/docs/Web/CSS/CSS_counter_styles). If omitted, the counter-style defaults to `decimal`.

> > **Note:** See also the {{cssxref("counters", "counters()")}} function which provides a third parameter, a string, to join the counter values when nesting counters.

Reviewer: This sentence can be flipped to say "To join..", "see..."

> > **Note:** See also the {{cssxref("counters", "counters()")}} function which provides a third parameter, a string, to join the counter values when nesting counters.

Author: ```suggestion
> **Note:** To join the counter values when nesting counters, use the {{cssxref("counters", "counters()")}} function, which provides an additional {{cssxref("string")}} parameter.
```

#### CSS

```css-nolint
ol {
  counter-reset: count;
}
li {
  counter-increment: count;
}
li::after {
  content:
    "[" counter(count, lower-roman) "] == ["
    counter(count, lower-alpha) "]";
}
```

#### Result

{{EmbedLiveSample("lower-roman compared to lower-alpha", "100%", 150)}}

> ### Counter displayed three ways

Reviewer: How about:
```suggestion
### Displaying a counter using three styles
```

#### CSS

> We include the `counter()` function with three different counter styles, including the default decimal value. We added padding to the list, providing space for the long `::marker` string.

Reviewer: ```suggestion
We include the `counter()` function with three different counter styles, including the default decimal value. We've added padding to the list to provide space for the long `::marker` string.
```
